polkadot/primitives/src/vstaging/mod.rs

```diff
@@ -143,17 +171,74 @@ impl<H> CandidateDescriptorV2<H> {
 		}
 	}
 
-	/// Set the PoV size in the descriptor. Only for tests.
-	#[cfg(feature = "test")]
-	pub fn set_pov_hash(&mut self, pov_hash: Hash) {
+	/// Check the signature of the collator within this descriptor.
+	pub fn check_collator_signature(&self) -> Result<(), ()> {
+		// Return `Ok` if collator signature is not included (v2+ descriptor).
+		let Some(collator) = self.collator() else { return Ok(()) };
+
+		let Some(signature) = self.signature() else { return Ok(()) };
+
+		super::v8::check_collator_signature(
+			&self.relay_parent,
+			&self.para_id,
+			&self.persisted_validation_data_hash,
+			&self.pov_hash,
+			&self.validation_code_hash,
+			&collator,
+			&signature,
+		)
+	}
+}
+
+/// A trait to allow changing the descriptor field values in tests.
+#[cfg(feature = "test")]
+
+pub trait MutateDescriptorV2<H> {
+	/// Set the relay parent of the descriptor.
+	fn set_relay_parent(&mut self, relay_parent: H);
+	/// Set the `ParaId` of the descriptor.
+	fn set_para_id(&mut self, para_id: Id);
+	/// Set the PoV hash of the descriptor.
+	fn set_pov_hash(&mut self, pov_hash: Hash);
+	/// Set the version field of the descriptor.
+	fn set_version(&mut self, version: InternalVersion);
+	/// Set the PVD of the descriptor.
+	fn set_persisted_validation_data_hash(&mut self, persisted_validation_data_hash: Hash);
+	/// Set the erasure root of the descriptor.
+	fn set_erasure_root(&mut self, erasure_root: Hash);
+	/// Set the para head of the descriptor.
+	fn set_para_head(&mut self, para_head: Hash);
+}
+
+#[cfg(feature = "test")]
```

Comment:
I'd rather directly add these methods onto the `CandidateDescriptorV2` and guard each of them with #[cfg(test)], which would save us the import of `MutateDescriptorV2`. But fine either way really